tests/Builder/Projection/SliceOperatorTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder\Projection;
+
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Projection;
+use MongoDB\Builder\Stage;
+use MongoDB\Tests\Builder\PipelineTestCase;
+
+/**
+ * Test $slice projection
+ */
+class SliceOperatorTest extends PipelineTestCase
+{
+    public function testReturnAnArrayWith3ElementsAfterSkippingTheFirstElement(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::project(
+                comments: Projection::slice(1, 3),
+            ),
+        );
+
+        $this->assertSamePipeline(Pipelines::SliceReturnAnArrayWith3ElementsAfterSkippingTheFirstElement, $pipeline);
+    }
+
+    public function testReturnAnArrayWith3ElementsAfterSkippingTheLastElement(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::project(
+                comments: Projection::slice(-1, 3),
```

Comment:
I think the argument order for `Projection::slice` will need to change, and I can see this being confusing for users since it will differ from the server API.

Both the short and long forms accept a `<number to return>` value, and the long form also accepts `<number to skip>`. If the PHP API is going to provide a consistent API without using named parameters, `<number to skip>` should be an optional second parameter.